cf/test/test_Data.py

```diff
@@ -2545,6 +2541,11 @@ def test_Data_percentile_median(self):
                         self.assertEqual(b1.shape, a1.shape)
                         self.assertTrue((b1.array == a1).all())
 
+        # Check for no warning when data is of masked type but with no
+        # missing values
+        d = cf.Data(np.ma.arange(100))
+        d.percentile(ranks[0])
```

Comment:
I guess from the comment you mean to inspect for no warning being raised by eye, but we can catch this as part of the test itself via the `warning` module:

```suggestion
        with warnings.catch_warnings():
            warnings.simplefilter("error", category=UserWarning)
            d = cf.Data(np.ma.arange(100))
            d.percentile(ranks[0])
```

(The docs [here](https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests) confirm this is the advised way to check no warnings are raised for the code lines in the context manager. Note FYI we also use `catch_warning` checking in the `test_basic_aggregate` though in that case to ignore warnings.)